LICENSE

```diff
@@ -0,0 +1,674 @@
+                    GNU GENERAL PUBLIC LICENSE
```

Comment:
is a license needed here?

Reply:
I put it in there to clarify the license of the repository itself. I can remove it if it doesn't fit though